components/director/cmd/director/main.go

```diff
@@ -468,3 +503,19 @@ func webhookService() webhook.WebhookService {
 
 	return webhook.NewService(webhookRepo, uidSvc)
 }
+
+func buildScheduler(namespace string, timeout time.Duration) (operation.Scheduler, error) {
+	cfg, err := rest.InClusterConfig()
```

Comment:
btw while testing the combined scheduler + controller flow I had to change this to the following in order to run the  director on localhost:

```
	var kubeconfig *string
	if home := homedir.HomeDir(); home != "" {
		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
	} else {
		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
	}
	flag.Parse()

	// use the current context in kubeconfig
	cfg, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
	if err != nil {
		panic(err.Error())
	}
```

Maybe it makes to introduce an env var switch which will be enabled when using `run.sh` so we can develop with local clusters as well.

Reply:
Reference links to running a k8s client in out-of-cluster mode:
1. https://github.com/kubernetes/client-go#how-to-use-it
2. https://github.com/kubernetes/client-go/tree/master/examples/out-of-cluster-client-configuration